File: src/algorithms/HouseRobber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        int *d = new int[n];
        int ans = 0;
        for (int i = 0; i < n; i++) {
            if (i == 0) d[i] = nums[i];
            else {
                int m = 0;
                for (int j = 0; j < i - 1; j++)
                    m = max(m, d[j]);
                d[i] = m + nums[i];
            }
            ans = max(ans, d[i]);
        }
        delete d;
        d = NULL;

        return ans;
    }
};

File: Algorithms/python/198-HouseRobber/HouseRobber.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        if len(nums) == 0:
            return 0
        d = list()
        d.append(nums[0])
        ans = nums[0]

        for i in range(1, len(nums)):
            t = 0
            for j in range(0, i - 1):
                t = max(t, d[j])
            a = t + nums[i]
            ans = max(ans, a)
            d.append(a)

        return ans
